cmd/producer/main.go

```diff
@@ -65,7 +65,7 @@ func main() {
 		}
 
 		// Send bytes to Output channel, after calling Initialise just in case it is not initialised.
-		kafkaProducer.Initialise(ctx)
+		_ = kafkaProducer.Initialise(ctx)
```

Comment:
The error is not being handled here. You should have something like this

```suggestion
		err = kafkaProducer.Initialise(ctx)
		if err != nil {
		        log.Fatal(ctx, "failed to initialise kafka producer", err)
	                os.Exit(1)
	        }
```